packages/next-codemod/bin/upgrade.ts

```diff
@@ -546,16 +547,31 @@ function writeOverridesField(
       if (!packageJson.pnpm.overrides) {
         packageJson.pnpm.overrides = {}
       }
-      for (const [key, value] of Object.entries(overrides)) {
+      for (const [key, value] of entries) {
         packageJson.pnpm.overrides[key] = value
       }
     }
   } else if (packageManager === 'yarn') {
     if (!packageJson.resolutions) {
       packageJson.resolutions = {}
     }
-    for (const [key, value] of Object.entries(overrides)) {
+    for (const [key, value] of entries) {
       packageJson.resolutions[key] = value
     }
+  } else if (packageManager === 'bun') {
+    // bun supports both bun.overrides and bun.resolutions
+    if (packageJson.resolutions) {
+      for (const [key, value] of entries) {
+        packageJson.resolutions[key] = value
+      }
+    } else {
+      // add overrides field if it's missing and add overrides
+      if (!packageJson.overrides) {
+        packageJson.overrides = {}
+      }
+      for (const [key, value] of entries) {
+        packageJson.overrides[key] = value
+      }
+    }
```

Comment:
You say `bun.resolutions` and `bun.overrides` but then check for top level `overrides` and `resolutions`. I'd include a link to docs in the PR description to make it clearer.

Reply:
wrong comment, updated